src/librbd/internal.cc

```diff
@@ -2273,11 +2279,16 @@ namespace librbd {
   void close_image(ImageCtx *ictx)
   {
     ldout(ictx->cct, 20) << "close_image " << ictx << dendl;
+    if (ictx->copyup_finisher != NULL) {
```

Comment:
Yes -- probably should move all copyup shutdown code to after everything else is flushed / shutdown.